pyproject.toml

```diff
@@ -13,11 +13,17 @@ maintainers = [
     {name = 'Caleb Evans', email = 'caleb@calebevans.me'}
 ]
 dependencies = [
-    "pydot>=1.4.2",
     "networkx>=2.6.2",
-    "frozendict>=2.3.4", #TODO I think that typing extensions needs to be in here
+    "frozendict>=2.3.4",
+    "typing-extensions>=4.5.0"
 ]
 
+# Per https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies
+[project.optional-dependencies]
+visual = ["coloraide>=1.8.2", "pygraphviz>=1.10"]
+
+#TODO maybe add stuff from here?: https://blog.whtsky.me/tech/2021/dont-forget-py.typed-for-your-typed-python-package/#adding-pytyped
```

Comment:
@eliotwrobson @khoda81 Just pointing out that per this TODO, we will still want to add `py.typed` per the linked article so that the user's mypy will be able to use our type annotations.

Reply:
Is `py.typed` included by default or do we have to set something in here? The file is already in the base directory on the `develop` branch.

Reply:
@eliotwrobson Hmm, a quick Google search on my end makes me think yes, it needs to be included in pyproject.toml as well ([source](https://stackoverflow.com/a/76214408/560642)):

```toml
[tool.setuptools.package-data]
"pkgname" = ["py.typed"]
```

And [here's an example](https://github.com/lsst/utils/blob/main/pyproject.toml#L50) of a project using the above in the wild.

Reply:
@caleb531 Added this here https://github.com/caleb531/automata/pull/129/commits/b0ce2adaf2f11bc39ea3568e683853e2240f3a03

In that same commit, I deleted the `.coveragerc` file and moved the configuration to the `pyproject.toml`. If you look at the comment there, I think the coverage issue has to do with the lines that get ignored.

Reply:
@eliotwrobson Ohh, I think I know what's going on.

I discovered with one of my recent projects that the `...` exclusion (which was originally added because `@overload` lines weren't getting covered for `DFA._cross_product`) were causing coverage weirdness.

You added the `@overload` decorators back in #131, but they were simplified and ultimately removed in #139. Therefore, it should be safe to remove `...` from the `exclude_lines` array.

P.S. Agreed about moving to `.coveragerc` to `pyproject.toml`, since it makes the coverage config the same format as the other config.

Reply:
@caleb531 I think that did it, removed the exclusion of the `...` here https://github.com/caleb531/automata/pull/129/commits/22317ac972e19b9499fa7dde0891bb6c2f0282a5

Reply:
@eliotwrobson Okay great! Also, regarding the exposing of types via `py.typed`, one way you could test this is to create a new Python project/virtualenv and use `pip install -e` to install your local clone of automata-lib into it. Then open your editor and see if the type checking with mypy is there (will probably need to install mypy as well, and make sure it is enabled in your editor/workspace).

Reply:
@eliotwrobson Just following up—did you ever get a chance to verify the functioning of `py.typed`?

Reply:
I think this is all working now. We'll want to test one more time before a release but I think it should be fine. I'll go ahead and resolve the discussion + delete some of the TODO comments in this file (as they have been resolved).